packages/env/lib/commands/run.js

```diff
@@ -53,6 +54,8 @@ function spawnCommandDirectly( { container, command, config, spinner } ) {
 		'-f',
 		config.dockerComposeConfigPath,
 		'run',
+		'-u',
```

Comment:
So I think this just changes the user when running something like `npx wp-env run wordpress bash`. I don't think this change would impact permissions as wp-env starts up. 🤔 

Another change I was just working on in #42826 will use a git checkout for the main WordPress version rather than whatever Docker provides. I wonder if that would help the permissions issue as well...

I also experimented with file permissions in #40864.

But I was unfortunately never able to replicate the bug (even on Linux!), so I'm not sure what the best path is forwards! (I'm also curious if the steps on this page helped at all with permissions issues: https://docs.docker.com/engine/install/linux-postinstall/)

Could you describe the issues you're seeing?

Reply:
There is some scripting involved, so it's hard to get an accurate step-by-step, but it goes something like this:

- wp-env start
- wp-env run tests-cli "wp theme install twentynineteen --activate"
 - wp-env destroy

File permission issues.

Even though, I could not reproduce it neither running these commands in a clean wp-env (this repo).

I'm not sure when the permission errors kicks-in, as it shouldn't happen if the `destroy` also happens inside the container. But if the `destroy` calls` `rm -rf` on the host, then it should have permission errors.

This is the error that I can reproduce when using wp-env integrated with our tooling:

![image](https://user-images.githubusercontent.com/9341686/182258825-96ec3e22-474e-460d-9e07-362b852f62f8.png)

Is that call to `rm -rf plugins` happening on the host machine? If so, that's a likely cause of permission issues

Reply:
Ops, my bad. `rm -rf plugins` is called on our end of the tooling.

This issue could theoretically be closed as a non-issue, but it's nice to have the UID mapping being done correctly in Linux as well.

Reply:
I've seen the #40864, I believe that mapping the host UID with the UID running inside the docker container is the most appopriate solution, as per this PR.

I believe that adding this UID mapping behind a feature flag, and optionally providing the ability to override Docker's UID with a environment variable will be enough to resolve the permission issues, as someone could do this if needed:

```
WP_ENV_UID=$(id -u):$(id -g) ./bin/wp-env run tests-cli "wp plugin download woocommerce" ("woocommerce" files will have correct permissions between host and Docker)
```

The Docker Desktop VM for Mac and Windows does something similar to this behind the scenes, but it's automatically managed for you, in Linux it has to be explicit. But if the `docker-compose run` call is behind an abstraction, such as wp-env's one, then the abstraction should allow to passthrough these arguments, ideally.

Reply:
A cleaner approach would be to add support for a `-u` or `--user` flag, mimicking docker-compose

Reply:
Good points! I also like the idea of keeping the user information consistent.

The reason I'm not sure this PR completely solves it is because when wp-env starts up, it's running docker compose commands directly, without using the built-in run command. The run command modified in this PR only changes the user settings for `npx wp-env run ...`. But when everything is installed with `npx wp-env start`, it's still using the default settings for everything.